portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,193 @@
+
+// Perform necessary setup
+document.addEventListener('DOMContentLoaded', () => {
+  if (documentHasElement(COMMENTS_CONTAINER)) {
+    // Retrieve comments data from the servlet and add to DOM
+    retrieveComments();
+  }
+});
+
+/** Adds the given comment to the page with the given ID. */
+function addSingleComment(comment) {
+  let containerId = COMMENT_CONTAINER_PREFIX + comment.id;
+
+  appendElement(COMMENTS_CONTAINER, 'div', '', /* id = */ containerId,
+                /* onClick = */ undefined, /* class = */ COMMENT_CLASS);
+  // Format the time nicely, by starting at the epoch and setting the milliseconds
+  let date = new Date(0);
+  date.setUTCMilliseconds(comment.timestamp);
+
+  // ID for the container which houses the information for a comment
+  let contentId = containerId + '-content';
+  appendElement(containerId, 'div', '', contentId, undefined, COMMENT_TEXT_CLASS);
+  // Add the name, time, and content
+  appendElement(contentId, 'p', `<b>${comment.name}</b>\t${date.toLocaleTimeString()}`);
+  appendElement(contentId, 'p', comment.text);
+
+  // Add the element that will toggle deleting this comment
+  appendElement(containerId, 'div', '', undefined, () => prepareDelete(comment.id),
+    COMMENT_SELECT_CLASS);
+}
+
+/**Adds the given list of comments to the page, and only adds them to the pageComments
+ * array if pushComments is true. startIndex indicates where to start counting up comment ID's */
+function addComments(comments, pushComments = true) {
+  comments.forEach(comment => {
+    addSingleComment(comment);
+
+    if (pushComments) {
+      pageComments.push(comment);
+    }
+    if (commentsToDelete.has(comment.id)) {
+      addClass(COMMENT_CONTAINER_PREFIX + comment.id, COMMENT_DELETE_CLASS);
+    }
+  });
+}
+
+/** Reverses the order of comments */
+function reverseComments() {
+  pageComments.reverse();
+  // Remove existing comments and add the reversed list
+  deleteChildren(COMMENTS_CONTAINER);
+  addComments(pageComments, /* pushComments = */ false, /* startIndex = */ 0);
+}
+
+/** Called when one of the sorting icons is picked. It changes the sort direction, if necessary */
+function commentSort(sortDirection) {
+  // Only change if the sorting direction has changed
+  if (sortDirection !== commentsSort) {
+    // Remove the ID (removes coloring) from the current sort
+    document.getElementById(COMMENTS_SORT_ICON).id = '';
+    commentsSort = sortDirection;
+
+    if (sortDirection === COMMENTS_SORT_NEWEST) {
+      // Add the ID (color) to the icon corresponding to newest
+      document.getElementsByClassName(COMMENTS_ICON_NEWEST)[0].id = COMMENTS_SORT_ICON;
+      reverseComments()
+    } else if (sortDirection === COMMENTS_SORT_OLDEST) {
+      // Add the ID (color) to the icon corresponding to oldest
+      document.getElementsByClassName(COMMENTS_ICON_OLDEST)[0].id = COMMENTS_SORT_ICON;
+      reverseComments();
+    }
+
+    // Finally, re-update the comments section
+    retrieveComments();
```

Comment:
I'm not sure why sorting happens and then comments are retrieved?

could you add some explanation to this?

Reply:
It's a relic of older code that I forgot to remove. Before I only reversed the comments without sending another GET request to the server. When I changed sorting to send another request, I forgot to remove those calls to `reverseComments()`. 
They have since been removed, but the function remains because I plan on using it in the future as it's not always necessary to send another HTTP request.